internal/app/appauth/storage.go

```diff
@@ -0,0 +1,347 @@
+	s.lock.Lock()
+	defer s.lock.Unlock()
+
+	delete(s.clients, requestID)
+	return nil
+}
+
+func (s *Storage) CreateAccessToken(_ context.Context, _ op.TokenRequest) (string, time.Time, error) {
+	return "id", time.Now().UTC().Add(5 * time.Hour), nil
+}
+
+func (s *Storage) CreateAccessAndRefreshTokens(_ context.Context, request op.TokenRequest, _ string) (accessTokenID string, newRefreshToken string, expiration time.Time, err error) {
+	authReq := request.(*AuthRequest)
+	return "id", authReq.ID, time.Now().UTC().Add(5 * time.Minute), nil
+}
+
+func (s *Storage) TokenRequestByRefreshToken(ctx context.Context, refreshToken string) (op.RefreshTokenRequest, error) {
+	r, err := s.AuthRequestByID(ctx, refreshToken)
+	return r.(op.RefreshTokenRequest), err
+}
+
+func (s *Storage) TerminateSession(_ context.Context, _, _ string) error {
+	return errors.New("not implemented")
+}
+
+func (s *Storage) GetSigningKey(_ context.Context, keyCh chan<- jose.SigningKey) {
+	keyCh <- jose.SigningKey{Algorithm: jose.RS256, Key: s.key}
+}
+
+func (s *Storage) GetKeySet(_ context.Context) (*jose.JSONWebKeySet, error) {
+	return &s.keySet, nil
+}
+
+func (s *Storage) GetKeyByIDAndUserID(_ context.Context, kid, _ string) (*jose.JSONWebKey, error) {
+	return &s.keySet.Key(kid)[0], nil
+}
+
+func (s *Storage) GetClientByClientID(_ context.Context, clientID string) (op.Client, error) {
+
+	if clientID == "" {
+		return nil, errors.New("invalid client id")
+	}
+
+	client, exists := s.clients[clientID]
+	if !exists {
+		return nil, errors.New("not found")
+	}
+
+	return client, nil
+
+	/*if id == "web" {
+		appType = op.ApplicationTypeWeb
+		authMethod = oidc.AuthMethodBasic
+		accessTokenType = op.AccessTokenTypeBearer
+		responseTypes = []oidc.ResponseType{oidc.ResponseTypeCode}
+	} else if id == "native" {
+		appType = op.ApplicationTypeNative
+		authMethod = oidc.AuthMethodNone
+		accessTokenType = op.AccessTokenTypeBearer
+		responseTypes = []oidc.ResponseType{oidc.ResponseTypeCode}
+	} else {
+		appType = op.ApplicationTypeUserAgent
+		authMethod = oidc.AuthMethodNone
+		accessTokenType = op.AccessTokenTypeJWT
+		responseTypes = []oidc.ResponseType{oidc.ResponseTypeIDToken, oidc.ResponseTypeIDTokenOnly}
+	}
+
+	return &ConfClient{ID: id, applicationType: appType, authMethod: authMethod, accessTokenType: accessTokenType, responseTypes: responseTypes, devMode: true}, nil */
+}
+
+func (s *Storage) AuthorizeClientIDSecret(_ context.Context, _ string, _ string) error {
+	return nil
+}
+
+func (s *Storage) SetUserinfoFromToken(ctx context.Context, userinfo oidc.UserInfoSetter, _, _, _ string) error {
+	return s.SetUserinfoFromScopes(ctx, userinfo, "", "", []string{})
+}
+
+func (s *Storage) SetUserinfoFromScopes(ctx context.Context, userinfo oidc.UserInfoSetter, subject, _ string, _ []string) error {
+
+	request, err := s.AuthRequestBySubject(ctx, subject)
+	if err != nil {
+		return err
+	}
+
+	userinfo.SetSubject(request.GetSubject())
```

Comment:
```suggestion
	// TODO: set user info correctly
	userinfo.SetSubject(request.GetSubject())
```

Implement, or mark with a comment